deps/github.com/arangodb/arangosync/client/api.go

```diff
@@ -0,0 +1,333 @@
+}
+
+//
+// TLSAuthentication contains configuration for using client certificates
+// and TLS verification of the server.
+type TLSAuthentication = tasks.TLSAuthentication
+
+type SynchronizationRequest struct {
+	// Endpoint of sync master of the source cluster
+	Source Endpoint `json:"source"`
+	// Authentication of the master
+	Authentication TLSAuthentication `json:"authentication"`
+}
+
+// Clone returns a deep copy of the given request.
+func (r SynchronizationRequest) Clone() SynchronizationRequest {
+	c := r
+	c.Source = r.Source.Clone()
+	return c
+}
+
+// IsSame returns true if both requests contain the same values.
+// The source is considered the same is the intersection of existing & given source is not empty.
+// We consider an intersection because:
+// - Servers can be down, resulting in a temporary missing endpoint
+// - Customer can specify only 1 of all servers
+func (r SynchronizationRequest) IsSame(other SynchronizationRequest) bool {
+	if r.Source.Intersection(other.Source).IsEmpty() {
+		return false
+	}
+	if r.Authentication.ClientCertificate != other.Authentication.ClientCertificate {
+		return false
+	}
+	if r.Authentication.ClientKey != other.Authentication.ClientKey {
+		return false
+	}
+	if r.Authentication.CACertificate != other.Authentication.CACertificate {
+		return false
+	}
+	return true
+}
+
+// Validate checks the values of the given request and returns an error
+// in case of improper values.
+// Returns nil on success.
+func (r SynchronizationRequest) Validate() error {
+	if len(r.Source) == 0 {
+		return errors.Wrap(BadRequestError, "source missing")
+	}
+	if err := r.Source.Validate(); err != nil {
+		return errors.Wrapf(BadRequestError, "Invalid source: %s", err.Error())
+	}
+	if r.Authentication.ClientCertificate == "" {
+		return errors.Wrap(BadRequestError, "clientCertificate missing")
+	}
+	if r.Authentication.ClientKey == "" {
+		return errors.Wrap(BadRequestError, "clientKey missing")
+	}
+	if r.Authentication.CACertificate == "" {
+		return errors.Wrap(BadRequestError, "caCertificate missing")
+	}
+	return nil
+}
+
+type CancelSynchronizationRequest struct {
+	// WaitTimeout is the amount of time the cancel function will wait
+	// until the synchronization has reached an `inactive` state.
+	// If this value is zero, the cancel function will only switch to the canceling state
+	// but not wait until the `inactive` state is reached.
+	WaitTimeout time.Duration `json:"wait_timeout,omitempty"`
+	// Force is set if you want to end the synchronization even if the source
+	// master cannot be reached.
+	Force bool `json:"force,omitempty"`
+	// ForceTimeout is the amount of time the syncmaster tries to contact
+	// the source master to notify it about cancelling the synchronization.
+	// This fields is only used when Force is true.
+	ForceTimeout time.Duration `json:"force_timeout,omitempty"`
+}
+
+type CancelSynchronizationResponse struct {
```

Comment:
The declaration makes it seem that it is ok for this struct to return an empty Json object.  Is that really acceptable?